### ISSUE TITLE
Deprecation around CFNetwork

### ISSUE THREAD

**Issue by diegostamigni** [tumtumtum/StreamingKit#424]
Some of the API in CFNetwork have been deprecated since macOS 10.11+. We can sort this out by replacing them accordingly with the ones from NSURLSession.

**Comment by diegostamigni** [comment]
Also duplicates of #266

**Comment by tumtumtum** [comment]
IIRC `NSURLConnection` has its own issues hence why I went down the lower level path. Tread carefully?

**Comment by diegostamigni** [comment]
yeah @tumtumtum ... thing is, all these Xcode warnings suggest to replace those CFNetwork constants with NSURLSession:
- [kCFStreamPropertyHTTPResponseHeader](https://developer.apple.com/documentation/cfnetwork/kcfstreampropertyhttpresponseheader?language=objc)
- [CFReadStreamCreateForHTTPRequest](https://developer.apple.com/documentation/cfnetwork/1426845-cfreadstreamcreateforhttprequest?language=objc)
- [kCFStreamPropertyHTTPShouldAutoredirect](https://developer.apple.com/documentation/cfnetwork/kcfstreampropertyhttpshouldautoredirect?language=objc)
- [kCFStreamPropertyHTTPProxy](https://developer.apple.com/documentation/cfnetwork/kcfstreampropertyhttpproxy?language=objc)
but really there isn't a replacement hence I was thinking of rewriting the whole `STKHTTPDataSource` to make usage of NSURLSession instead (with back compatibility [probably just an additional data source and swap based on SDK/runtime version, similar to what I've done in #423 for OSSpinLock]) though I'm not sure we can read stream the way we do with CFNetwork 🤔 (haven't look at it yet).
